nodebuilder/state/core.go

```diff
@@ -30,14 +33,31 @@ func coreAccessor(
 	*modfraud.ServiceBreaker[*state.CoreAccessor, *header.ExtendedHeader],
 	error,
 ) {
-	ca, err := state.NewCoreAccessor(keyring, string(keyname), sync, corecfg.IP, corecfg.GRPCPort,
-		network.String(), opts...)
+	if corecfg.TLSEnabled {
+		tlsCfg, err := core.TLS(corecfg.TLSPath)
+		switch {
+		case err == nil:
+		case errors.Is(err, os.ErrNotExist):
+			// set an empty config if path is empty under `TLSEnabled=true`
+			tlsCfg = core.EmptyTLSConfig()
+		default:
+			return nil, nil, nil, err
+		}
+
+		xtoken, err := core.XToken(corecfg.XTokenPath)
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return nil, nil, nil, err
+		}
+		opts = append(opts, state.WithTLSConfig(tlsCfg), state.WithXToken(xtoken))
```

Comment:
https://github.com/celestiaorg/celestia-node/pull/3922#discussion_r1850192010

Reply:
So even if we have empty string for xtoken here, we still set state.WithXToken?

Reply:
I guess it's fine bc it's implicitly handled inside core access when it sets up the gRPC but it's still odd. It's fine tho just a nit, feel free to resolve.

Reply:
> So even if we have empty string for xtoken here, we still set state.WithXToken?

It will set an empty string, which will be set anyway in core_accessor if we don't pass anything.